sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StatefulProcessorHandleImpl.scala

```diff
@@ -103,22 +113,29 @@ class StatefulProcessorHandleImpl(
 
   private var currState: StatefulProcessorHandleState = CREATED
 
-  private def verify(condition: => Boolean, msg: String): Unit = {
-    if (!condition) {
-      throw new IllegalStateException(msg)
-    }
-  }
-
   def setHandleState(newState: StatefulProcessorHandleState): Unit = {
     currState = newState
   }
 
   def getHandleState: StatefulProcessorHandleState = currState
 
-  override def getValueState[T](stateName: String, valEncoder: Encoder[T]): ValueState[T] = {
+  override def getValueState[T](
+      stateName: String,
+      valEncoder: Encoder[T]): ValueState[T] = {
     verifyStateVarOperations("get_value_state")
-    val resultState = new ValueStateImpl[T](store, stateName, keyEncoder, valEncoder)
-    resultState
+
+    if (ttlMode == TTLMode.NoTTL()) {
+      new ValueStateImpl[T](store, stateName, keyEncoder, valEncoder)
+    } else {
+      val valueStateWithTTL = new ValueStateImplWithTTL[T](store, stateName,
+        keyEncoder, valEncoder, ttlMode, batchTimestampMs, eventTimeWatermarkMs)
+
+      val ttlState = valueStateWithTTL.ttlState
```

Comment:
Not sure what we want to do here ? could we just add the `valueStateWithTTL` instance to the list of `ttlStates` ?

Reply:
Maybe I am not following your question. We want to add the ttlState instead of the actual state variable state to `ttlStates`.